plugins/org.eclipse.n4js.generator.headless/src/org/eclipse/n4js/generator/headless/N4HeadlessCompiler.java

```diff
@@ -99,13 +99,27 @@
  *
  * The way how the compiler behaves can be configured through flags like {@link #keepOnCompiling},
  * {@link #processTestCode}, {@link #compileSourceCode}
+ *
+ * IMPORTANT: Before using the functionalities of this class, make sure to register composite generators first.
+ * Moreover, the subgenerators must have been registered as well.
  */
 public class N4HeadlessCompiler {
 
-	/** The Generator to compile with */
-	private final CompositeGenerator compositeGenerator;
+	/** The composite generator that manages all subgenerators. */
+	@Inject
+	private ICompositeGenerator compositeGenerator;
+
+	/** Get composite generator. */
+	public ICompositeGenerator getCompositeGenerator() {
+		return compositeGenerator;
+	}
+
+	/** Set composite generator. */
+	public void setCompositeGenerator(ICompositeGenerator compositeGenerator) {
+		this.compositeGenerator = compositeGenerator;
```

Comment:
in the above 10 lines you both inject the composite generator AND set it with a setter. That seems redundant.

Reply:
Removed. Done.